# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1234793520 -3600
#      Mon Feb 16 15:12:00 2009 +0100
# Node ID 1db8033efa17bf3099fcd02a3ea0d7101c7c2e49
# Parent  b0356f30a33b707777704b307a309248577fd725
fix which message for built-in functions

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,11 +1,12 @@
 2009-02-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* linear-algebra/expm.m: Fix invalid unscaling.
+	* help/which.m: Do not add "function" here.
 
 2009-02-16  John W. Eaton  <jwe@octave.org>
 
 	* help/help.m: Shorten default help text.
 
 2009-02-15  John W. Eaton  <jwe@octave.org>
 
 	* help/doc.m, help/help.m, help/lookfor.m, help/type.m,
diff --git a/scripts/help/which.m b/scripts/help/which.m
--- a/scripts/help/which.m
+++ b/scripts/help/which.m
@@ -25,17 +25,17 @@ function varargout = which (varargin)
 
   if (nargin > 0 && iscellstr (varargin))
     m = __which__ (varargin{:});
 
     if (nargout == 0)
       for i = 1:nargin
 	if (isempty (m(i).file))
 	  if (! isempty (m(i).type))
-	    printf ("`%s' is a %s function\n",
+	    printf ("`%s' is a %s\n",
 		    m(i).name, m(i).type);
 	  endif
 	else
 	  if (isempty (m(i).type))
 	    printf ("`%s' is the file %s\n",
 		    m(i).name, m(i).file);
 	  else
 	    printf ("`%s' is a %s from the file %s\n",
